api/src/main/scala/com/gu/adapters/http/Authentication.scala

```diff
@@ -48,10 +81,69 @@ class AuthenticationService(idapiAuthService: IdapiAuthService) {
 
     // Convert authentication result to return type
     // (identity-auth-core uses cats (Either); discussion-avatar uses scalaz (\/)).
-    result.redeem(
-      err => -\/(userAuthorizationFailed(NonEmptyList(err.getMessage)): Error),
-      identityId => \/-(User(identityId))
-    ).unsafeRunSync()
+    result
+      .redeem(
+        err =>
+          -\/(userAuthorizationFailed(NonEmptyList(err.getMessage)): Error),
+        identityId => \/-(User(identityId))
+      )
+      .unsafeRunSync()
+  }
+
+  private def authenticateUserWithOkta(
+    accessToken: Option[String],
+    identityAccessScope: IdentityAccessScope
+  ): Error \/ User = {
+    // Attempt to authenticate user.
+    val result = for {
+      value <- IO.fromEither(
+        Either.fromOption(
+          accessToken,
+          new Exception("No oauth access token in request")
+        )
+      )
+      credentials = value.stripPrefix("Bearer ")
+      claims <- IO.fromEither(
+        oktaLocalValidator.parsedClaimsFromAccessToken(
+          credentials,
+          List(identityAccessScope),
+          DefaultAccessClaimsParser
+        ).leftMap(e => new Exception(e.message, new Exception(e.suggestedHttpResponseCode.toString)))
```

Comment:
I think you could call `oauthTokenAuthorizationFailed` directly here instead of converting an `Exception`:
```suggestion
        ).leftMap(e => oauthTokenAuthorizationFailed(NonEmptyList(e.getMessage), e. suggestedHttpResponseCode.toString)))
```

Reply:
![Screenshot 2023-06-15 at 15 24 35](https://github.com/guardian/discussion-avatar/assets/13315440/7d431b50-4f95-4248-b066-1bb1369730f3)

This didn't quite work, as the `oauthTokenAuthorizationFailed` doesn't extend the Throwable type

Reply:
Oh that's a pity.  I'll try checking out the branch to see if there's any potential for simplifying this bit